### PR TITLE
add XmlDatabaseTest.java with failing test for issue #297

### PR DESCRIPTION
**Pull request by ktarbet**
Draft PR - Documents  #297  with junit test.

I was able to re-work the test to create a database from scratch, and reproduce the issue.  No input files required.

### CODE REVIEW

**Comment by MikeNeilson** [comment]
The structure of the test looks great to me. Do you have time to try and fix the actual issue or do you need to pass the buck at this point?

**Comment by ktarbet** [comment]
I'll spend some time on the actual issue.

**Comment by ktarbet** [comment]
The potential solution in https://github.com/opendcs/opendcs/pull/305/commits/a38eb28e690dd0a1baa0aa50aaefd55743e07c7d 
uses lists of  DecodesScriptParser and DecodesScriptBuilder, in the PlatformConfigParser.  When the endTag </PlatformConfig> is reached everything is ready to parse in the endElement() method

File: src/main/java/decodes/xml/PlatformConfigParser.java [hunk 1]
```diff
@@ -24,8 +27,8 @@ public class PlatformConfigParser implements XmlObjectParser, XmlObjectWriter, T
 	/**
 	 * Reference to the DecodesScriptBuilder so we can finalized after we've read it in.
 	 */
-	private DecodesScriptBuilder scriptBuilder; 
-	private DecodesScriptParser scriptParser;
+	private List<DecodesScriptParser> decodesScriptParserList = new ArrayList<>();
+	private List<DecodesScriptBuilder> decodesScriptBuilderList = new ArrayList<>();
```
Comment (MikeNeilson) [inline src/main/java/decodes/xml/PlatformConfigParser.java:31]: This looks like a reasonable solution to me. Reading through it I suspect we could limit this to just an array of `DecodeScriptBuilder`s. I think my original work in this design it seemed impractical to do so.

But that can be addressed later.

**Review by MikeNeilson — APPROVED** [review]
This looks good to me.

However, I will leave it a day or two for others to review.

**Pull request merged**